### PR TITLE
[PLATFORM-578]: [Auth0Ex] Wrong error handling on cache decrypt failure

### PR DESCRIPTION
**Pull request by cpiemontese**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/PLATFORM-578

This PR tries to handle decryption errors better i.e. instead of returning `{:ok, :error}`, return `{:error, "Failed to decrypt token"}`

I also added a CODEOWNERS file to try to see if we like it

### CODE REVIEW

File: test/prima_auth0_ex/token_provider/token_encryptor_test.exs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    plaintext = "test"
+
+    {:ok, enc} = TokenEncryptor.encrypt(plaintext)
+
+    TestHelper.set_client_env(:cache_encryption_key, 1234, false)
+
+    assert {:error, _} = TokenEncryptor.decrypt(enc)
+  end
+
+  defp keygen, do: Base.encode64(:crypto.strong_rand_bytes(32))
```
Comment (Fs00) [inline test/prima_auth0_ex/token_provider/token_encryptor_test.exs:56]: Could this be called `generate_key` or something like that? I couldn't help but think to those tools used by 🏴‍☠️ 😂

File: test/test_helper.exs [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+  import ExUnit.Callbacks
+
+  @spec set_client_env(term(), term(), boolean()) :: :ok
+  def set_client_env(key, value, false) do
+    Application.get_env(:prima_auth0_ex, :client, [])
+    |> Keyword.put(key, value)
+    |> then(&Application.put_env(:prima_auth0_ex, :client, &1))
+  end
+
+  def set_client_env(key, value, true) do
```
Comment (Fs00) [inline test/test_helper.exs:14]: Could the boolean parameter be a keyword list instead, something like `reset_after_test: true`? By reading the test it isn't clear what it does.

Btw, these helpers are nice: can be used in other tests too?

File: test/prima_auth0_ex/token_provider/token_encryptor_test.exs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    {:ok, enc} = TokenEncryptor.encrypt(plaintext)
+    {:ok, dec} = TokenEncryptor.decrypt(enc)
+
+    assert plaintext == dec
+  end
+
+  test "decrypting returns with :error if key changes" do
+    TestHelper.set_client_env(:cache_encryption_key, @token_encryption_key, true)
+
+    plaintext = "test"
```
Comment (Fs00) [inline test/prima_auth0_ex/token_provider/token_encryptor_test.exs:25]: Can this variable be inlined in this specific test? The value isn't used afterwards.

File: test/prima_auth0_ex/token_provider/token_encryptor_test.exs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    assert {:error, _} = TokenEncryptor.encrypt(plaintext)
+  end
+
+  test "decrypting returns with :error if key is not binary" do
+    TestHelper.set_client_env(:cache_encryption_key, @token_encryption_key, true)
+
+    plaintext = "test"
```
Comment (Fs00) [inline test/prima_auth0_ex/token_provider/token_encryptor_test.exs:47]: Same as line 25

**Review by Fs00 — COMMENTED** [review]
The implementation looks fine and correct, I have only a few observations on tests

**Comment by cpiemontese** [comment]
I looked into using the helper for other tests, but they don't seem to be modifying the config

**Review by Fs00 — APPROVED** [review]

**Pull request merged**